Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt

```diff
@@ -396,117 +367,106 @@ class FacilityInfoPage : BaseActivity() {
     }
 
-    /**
-     * This function sets the operating hours text under the historical data chart
-     * @param day The day for which the operating hours is set
-     */
-    private fun setOperatingHoursText(day: String, date: Date) {
-        Log.d("SET", "OPERATING")
-        val hourTitle = FluxUtil.dayFullString(day)
-        todayHours.text = hourTitle
-        val todayDateFormat = SimpleDateFormat("MMMMM dd", Locale.US)
-        todayDate.text = todayDateFormat.format(date)
-        val todayDayDateFormat = SimpleDateFormat("EEE, MMM dd", Locale.US)
-        todayDayDate.text = todayDayDateFormat.format(date)
-        if (opHoursStrings.isEmpty()) facilityHours.text = getString(R.string.no_operating_hours)
-        else facilityHours.text = ""
-        for (operatingSegment in opHoursStrings) {
-            val allHours = facilityHours.text.toString() + operatingSegment + if (opHoursStrings.indexOf(operatingSegment) == opHoursStrings.size - 1) "" else "\n"
-            facilityHours.text = allHours
-        }
-    }
-
     override fun updateUI() {
         Log.d("updatedFPUI", "updating")
         fetchOperatingHours(date = FluxUtil.getDateObject(selectedDay!!)) // TODO FIX!! ON REFRESH!!
     }
 
-    private fun fetchHistoricalJSON(day: String, facilityId: String) {
-        api.fetchHistoricalJSON(
-                day = day,
-                facilityId = facilityId,
-                fetchHistoricalJSONOnResponse = { historicalDensities ->
-                    densities = historicalDensities
-                    setupBarChart()
-                }
-        )
-    }
-
     private fun fetchMenuJSON(day: String, facilityId: String) {
         api.fetchMenuJSON(
                 day = day,
                 facilityId = facilityId
         ) { menu ->
+            // set the current menu to the fetched menu
+            currentMenu = menu
+
+            // once loaded, hide the loader
             menuProgressBar.isGone = true
+
+            val dayAvailableMenus = mutableListOf<String>()
+            val dayAvailableMenuTabs = mutableListOf<TabLayout.Tab>()
+
+            if (menu?.breakfastItems?.size != 0) {
+                dayAvailableMenuTabs.add(menuTabs.newTab().setText(R.string.breakfast))
+                dayAvailableMenus.add("breakfast")
+            }
+            if (menu?.brunchItems?.size != 0) {
+                dayAvailableMenuTabs.add(menuTabs.newTab().setText(R.string.brunch))
+                dayAvailableMenus.add("brunch")
+            }
+            if (menu?.lunchItems?.size != 0) {
+                dayAvailableMenuTabs.add(menuTabs.newTab().setText(R.string.lunch))
+                dayAvailableMenus.add("lunch")
+            }
+            if (menu?.dinnerItems?.size != 0) {
+                dayAvailableMenuTabs.add(menuTabs.newTab().setText(R.string.dinner))
+                dayAvailableMenus.add("dinner")
+            }
+
             if (menu?.breakfastItems?.size == 0
```

Comment:
Yeah, I guess that is sort of possible. The reason it's like this currently is because `menu` and `breakfastItems` are nullable, so we need to call make a call/comparison with the `?` to check for null. I could fix it to be the following:

```
menu?.breakfastItems?.isEmpty() == true
```

This way it uses the `isEmpty()` function, but it can't be called without the `?` and the equality check.

Reply:
Hmm that's what I thought, but the next three lines use isEmpty():
```
if (menu?.breakfastItems?.size == 0
    && menu.brunchItems.isEmpty()
    && menu.lunchItems.isEmpty()
    && menu.dinnerItems.isEmpty())
``` 
which is why I was asking for consistency. Maybe those lines should be changed?

Reply:
I've updated it to use the following:

```
menu?.breakfastItems?.isEmpty() == true
```

It does look a little more consistent.

Reply:
Is `== true` redundant?

Reply:
Nope. Because `menu` and `breakfastItems` are nullable, calling `isEmpty()` returns a `Boolean?` type (nullable boolean - either `Boolean` or `null`). And, conditions only want `Boolean` types to evaluate, so we need to enforce that it's actually true.

Reply:
Interesting!